### DISCUSSION TITLE
Aurora JDBC Failover plugin with custom endpoints

### DISCUSSION THREAD

**Discussion by gagankalra1** [aws/aws-advanced-jdbc-wrapper#722] [Q&A]
IHAC who is using the Failover Plugin for the advanced JDBC wrapper. Question - whether the failover plugin works the same if we use custom endpoints, rather than standard cluster endpoints?

https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheFailoverPlugin.md

**Comment by brunos-bq** [comment]
Hello @gagankalra1,

The answer to your question is a bit more complicated than a simple 'yes' or 'no'. The AWS JDBC Driver does support custom endpoints, but at the end, it all depends on what is the user setup and what the user is trying to achieve.

When using a custom domain, one has to manually set up a host pattern, a regex expression to indicate the AWS JDBC Driver how to find the available hosts in the cluster and failover to them when necessary.

There is a section on the documentation that addresses host patterns: 
https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheFailoverPlugin.md#host-pattern

Please let us know if you have any other related questions.
